test/razorpay/test_payment.rb

```diff
@@ -79,5 +142,35 @@ def test_payment_capture_without_fetch
       payment = Razorpay::Payment.capture(@payment_id, amount: 5100)
       assert_equal 'captured', payment.status
     end
+
+    def test_payment_edit
+
+      payment_attr = {
+        "notes": {
+          "key1": 'value1',
+          "key2": 'value2'
+        }
+      } 
+      
+      stub_patch(%r{payments\/#{"pay_IY4ljejpy9KUr9"}$}, 'fake_update_payment', payment_attr.to_json)
+      payment = Razorpay::Payment.edit("pay_IY4ljejpy9KUr9",payment_attr.to_json)
+      assert_equal 'captured', payment.status
+    end
+
+    def test_payment_create_payment_json
+
+      payment_attr = {
+        amount: '100',
+        currency: 'INR',
+        email: 'gaurav.kumar@example.com',
+        contact: '9123456789',
+        order_id: 'order_EAkbvXiCJlwhHR',
+        method: 'card'
```

Comment:
Card details is missing - https://razorpay.com/docs/payments/payment-gateway/s2s-integration/json/v2/#step-2-create-a-payment

Pls see you are also simulating for other methods as well.